### DISCUSSION TITLE
wfb-cli drone [statistics disabled]

### DISCUSSION THREAD

**Discussion by 1113870631** [svpcom/wfb-ng#212] [Questions] [answered]
i use the wfb-cli drone command shows that 
![2022-04-11 12-18-52 的屏幕截图](https://user-images.githubusercontent.com/61228813/162664401-4c976f3a-be81-41ee-bfe4-9c7e820a7a13.png)
is it ok?

there is log 
![2022-04-11 12-20-30 的屏幕截图](https://user-images.githubusercontent.com/61228813/162664489-1de3e4d4-40a5-4de9-afab-b72206b97996.png)
on my gs the link is lost
![2022-04-11 12-21-14 的屏幕截图](https://user-images.githubusercontent.com/61228813/162664539-cf653234-8a71-4e63-928f-16e2f7b4a581.png)
there is log
![2022-04-11 12-22-02 的屏幕截图](https://user-images.githubusercontent.com/61228813/162664627-eb507c6d-17e3-4135-92fc-e0068c966ef4.png)
![2022-04-11 12-22-21 的屏幕截图](https://user-images.githubusercontent.com/61228813/162664643-c017eb8f-6386-4e54-a212-9f5a112f6acf.png)

**Comment by svpcom** [comment, marked as answer]
Yes. By default statistics collection is disabled on the drone side. To enable it add to /etc/wifibroadcast.cfg on the drone:
```
[drone_mavlink]
stats_port = 8001

[drone_video]
stats_port = 8002

[drone_tunnel]
stats_port = 8003
```

**Reply by 1113870631** [reply]
thanks i add this .the status is showed ok ,but why i can't connect the gs. the video and tunnel is link lost,
the drone is pi4B the gs is my computer with the rlt8812au
this is drone status and log 
![2022-04-11 13-55-40 的屏幕截图](https://user-images.githubusercontent.com/61228813/162673526-e593d2f6-08ba-4711-8640-16930185f4b2.png)
![2022-04-11 13-56-59 的屏幕截图](https://user-images.githubusercontent.com/61228813/162673654-876d62bd-25e8-41a7-a674-d77f74dde112.png)

i have tested it on my computer use two rtl8812au ,it's ok
but on my pi4B  is not ok

**Comment by svpcom** [comment]
You need to run `wfb-cli gs` on the GS side

**Reply by svpcom** [reply]
You have different wifi channels configured: 1 vs 161

**Reply by svpcom** [reply]
Set common channel number

**Reply by svpcom** [reply]
On the drone you can use `peer = 'serial:ttyUSB0:115200'` to avoid use of mavlink_router

**Reply by svpcom** [reply]
See updated SetupHOWTO

**Reply by 1113870631** [reply]
ok thanks it's connected and on the drone i use the ffmpeg to send a udp stream such as :
ffmpeg -f v4l2 -i /dev/video0 -f flv udp://127.0.0.1:5602
the video connect is ok the wfb-cli shows like this
![2022-04-11 14-44-10 的屏幕截图](https://user-images.githubusercontent.com/61228813/162679820-92b57d56-4c1a-4186-b43e-d0c10761abe8.png)
but on the gs  i use ffplay udp://127.0.0.1:5600   there is no video windows comes out 

is it the video stream no the rigitht  formate 
what format should i use  
and the wfb-cli gs shows the connect is not stable ,it  is link lost ，It happens every once in a while，is it normal?

**Comment by svpcom** [comment]
Do you use raspberry pi for drone? If yes, try prebuilt images https://github.com/svpcom/wifibroadcast/releases/download/wifibroadcast-21.08/wifibroadcast_22.2-beta2-rpi3.img.gz

To grab and display video use gstreamer pipelines is preferred.
```
# display stream on X11 from UDP 5600
gst-launch-1.0 udpsrc do-timestamp=true port=5600 caps='application/x-rtp' ! rtph264depay ! h264parse disable-passthrough=true ! avdec_h264 ! xvimagesink sync=false
```

```
# grab stream from UVC camera use software encoder to UDP 5602
gst-launch-1.0 v4l2src device=/dev/video0  ! videoconvert ! video/x-raw,format=NV12,framerate=30/1,width=640,height=360 ! x264enc tune=zerolatency bitrate=4000  ! rtph264pay mtu=1400 config-interval=1 ! udpsink host=127.0.0.1 port=5602 sync=false
```

**Reply by 1113870631** [reply]
I'm not familiar with GStreamer so i will try the prebuilt images
just flash the img to SD and reboot?

**Reply by 1113870631** [reply]
my pi is 4B can use the image?

**Reply by svpcom** [reply]
yes, flash and reboot.  I've test them with RPI 3b, so you can try them with RPI4

**Reply by 1113870631** [reply]
ok thanks

**Reply by 1113870631** [reply]
> yes, flash and reboot. I've test them with RPI 3b, so you can try them with RPI4

i am sorry ,i am back ,use this img i can't connect wifi and can't get into ssh

**Reply by svpcom** [reply]
wifi is unavailable. Use serial console or ssh via ethernet

**Reply by 1113870631** [reply]
ok ,but i just tested on my computer with two rtl8812au ,i  found that the if use gstreamer the video is clear ,the ffmpeg is bad(maybe my ffmpeg video stream is not right )but whatever i use the connection will be disconnected in a moment ，the link is lost ,would you know why ？

**Reply by svpcom** [reply]
your ffmpeg stream use "flv" which is invalid for WFB. You need to stream h264 encapsulated into RTP

**Reply by 1113870631** [reply]
I tested the ffmpeg -f v4l2 -i  /dev/video0 -f h264 udp://127.0.0.1:5602 is also bad ,the bigger question is it will be disconnected in a minute,can i get more detailed log to see what happened

**Reply by svpcom** [reply]
If it is disconnected in a minute then check that networkmanager and wpa_supplicant are properly configured according to howto

**Reply by 1113870631** [reply]
In detail, it will be reconnected intermittently a few times at first, and then completely disconnected and then i  restart the gs  program it will  continue the previous loop

**Reply by svpcom** [reply]
It seems that some other program (networkmanager, wpa_supplicant, dhclient, etc) touch network interface. Need to disable such programs

**Reply by 1113870631** [reply]
yes i kill the all wpa_supplicant it is ok now ,thank you very much ，I think I should write your name on my senior thesis，hahaha ,the last question can i use it to tranform Images at unconventional resolutions？ ag:1280x480（Binocular vision）

**Reply by svpcom** [reply]
yes. It doesn't depends what image is transferred (1920x1080 from single camera or 1280x480 combined from two cameras)

**Reply by 1113870631** [reply]
ok thank you very very very ..................................................very much

**Reply by svpcom** [reply]
Could you describe your setup and your project? I'll add it to success stories

**Reply by 1113870631** [reply]
sure I sorted it out
[github_wifibroad_cast.pdf](https://github.com/svpcom/wifibroadcast/files/8469759/github_wifibroad_cast.pdf)
[github_wifibroad_cast.odt](https://github.com/svpcom/wifibroadcast/files/8469761/github_wifibroad_cast.odt)

**Reply by 1113870631** [reply]
@svpcom did you get it

**Reply by svpcom** [reply]
yes, thanks!

**Reply by svpcom** [reply]
https://github.com/svpcom/wifibroadcast/wiki/Success-stories
